WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift

```diff
@@ -53,11 +54,47 @@ enum ProductSettingsRows {
         let cellTypes: [UITableViewCell.Type] = [BasicTableViewCell.self]
     }
 
+    struct Status: ProductSettingsRowMediator {
+        private let settings: ProductSettings
+
+        init(_ settings: ProductSettings) {
+            self.settings = settings
+        }
+
+        func configure(cell: UITableViewCell) {
+            guard let cell = cell as? SettingTitleAndValueTableViewCell else {
+                return
+            }
+
+            cell.updateUI(title: NSLocalizedString("Status", comment: "Status label in Product Settings"), value: settings.status.description)
+            cell.accessoryType = .disclosureIndicator
+        }
+
+        func handleTap(sourceViewController: UIViewController, onCompletion: @escaping (ProductSettings) -> Void) {
+            let title = NSLocalizedString("Status", comment: "Product status setting list selector navigation title")
+            let viewProperties = ListSelectorViewProperties(navigationBarTitle: title)
+            let dataSource = ProductStatusSettingListSelectorDataSource(selected: settings.status)
+
+            let listSelectorViewController = ListSelectorViewController(viewProperties: viewProperties,
+                                                                        dataSource: dataSource) { selected in
```

Comment:
If a try to use `[weak self]` I get an error, because 'weak' may only be applied to class and class-bound protocol types, but here we are inside a `struct`, where value types are assigned by copy. Reference counting does not apply, so the `weak` modifier does not make sense in this case 😉